File: java-implements/queue/LLQueue.java
public class LLQueue {
    
    public class Node{
        int data;
        Node next;

        Node(int d){
            data = d;
            next = null;
        }
    }

    public static Node front;
    public static Node rear;

    public void add(int data){
        Node newNode = new Node(data);
        if(front == null){
            rear = front = newNode;
            return;
        }
        rear.next = newNode;
        rear = newNode;
        return;
    }
    
    public int remove(){
        if(front == null){
            System.out.println("Queue is Empty!");
            return -1;
        }

        int val = front.data;
        front = front.next;
        return val;
    }

    public int peek(){
        if(front==null){
            System.out.println("Queue is Empty!");
            return -1;
        }
        return front.data;
    }

    public void printQueue(Node front){
        if(front==null){
            System.out.println("Queue is Empty!");
            return;
        }

        while(front!=null){
            System.out.print(front.data+" ");
            front = front.next;
        }
        
    }

    public static void main(String[] args) {
        LLQueue a = new LLQueue();
        a.add(1);
        a.add(2);
        a.add(3);
        a.remove();
        a.remove();
        a.remove();
        // a.printQueue();
        System.out.println(a.peek());        
    }
}

File: cpp-implements/heap/llqueue.cc
	#include <iostream>
	using namespace std;

	class Node{
	public:
		int val;
		Node* next;
	
		Node(int val) : val(val), next(nullptr) {}
	};

	class LLQueue{
	private:
		Node* head;
		Node* tail;

	public:
		LLQueue() : head(nullptr), tail(nullptr) {}

		~LLQueue() {
			while(head != nullptr){
				Node* tmp = head;
				head = head->next;
				delete tmp;
			}
			tail = nullptr;
		}

		bool isEmpty(){
			return head == nullptr;
		}

		void enqueue(int val){
			Node* newNode = new Node(val);

			if(isEmpty()){
				head = tail = newNode;
			} else{
				tail->next = newNode;
				tail = tail->next;
			}
		}

		int dequeue(){
			if(isEmpty()){
				cout<<"Queue is overflowing\n";
				return -1;
			}

			int val = head->val;
			Node* tmp = head;
			head = head->next;
			delete tmp;

			if(head==nullptr) tail=nullptr;
			return val;
		}

		int peek(){
			if(isEmpty()){
				cout<<"Queue is overflowing\n";
				return -1;
			}

			return head->val;
		}

	};


	int main(){
		LLQueue q;
	    q.enqueue(10);
	    q.enqueue(20);
	    q.enqueue(30);
	    cout << "Front element: " << q.peek() << endl;
	    cout << "Dequeue: " << q.dequeue() << endl;
	    cout << "Front element after dequeue: " << q.peek() << endl;

		return 0;
	}	